File: archive/p/python/longest_word.py
import sys

error = "Usage: please provide a string"

if len(sys.argv) == 1 or sys.argv[1] == "":
    print(error)
    sys.exit(1)

input_string = sys.argv[1]
longest_word = max(len(word) for word in input_string.split())
print(longest_word)

File: archive/p/php/longest-word.php
<?php
function usage()
{
    exit("Usage: please provide a string");
}

function longest_word_length($str)
{
    // Split on whitespace, get string length of each word, take the maximum
    return max(array_map("strlen", preg_split("/\s+/", $str)));
}

// Exit if too few arguments or 1st command-line argument is empty
if (count($argv) < 2 || empty($argv[1]))
{
    usage();
}

// Get longest word length and display
$str = $argv[1];
$longest_len = longest_word_length($str);
echo "${longest_len}\n";
